api/v1/services/product.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Optional
+from sqlalchemy.orm import Session
+
+from api.core.base.services import Service
+from api.utils.db_validators import check_model_existence
+from api.v1.models.product import Product
+
+
+class ProductService(Service):
+    '''Product service functionality'''
+
+    def create(self, db: Session,  schema):
+        '''Create a new product'''
+
+        new_product = Product(**schema.model_dump())
+        db.add(new_product)
+        db.commit()
+        db.refresh(new_product)
+
+        return new_product
+    
+
+    def fetch_all(self, db: Session, **query_params: Optional[Any]):
+        '''Fetch all products with option tto search using query parameters'''
+
+        query = db.query(Product)
+
+        # Enable filter by query parameter
+        if query_params:
+            for column, value in query_params.items():
+                if hasattr(Product, column) and value is not None:
```

Comment:
```suggestion
                if hasattr(Product, column) and value:
```

if obj checks that obj is not empty/null (None)